src/crimson/osd/osd.cc

```diff
@@ -608,6 +610,41 @@ seastar::future<> OSD::_send_boot()
   return monc->send_message(std::move(m));
 }
 
+seastar::future<> OSD::_add_device_class()
+{
+  LOG_PREFIX(OSD::_add_device_class);
+  if (!local_conf().get_val<bool>("osd_crush_update_on_start")) {
+    return seastar::now();
+  }
+
+  auto get_device_class = [this]() -> seastar::future<std::string> {
+    return store.get_default_device_class().then([FNAME, this](std::string device_class) {
+      return seastar::make_ready_future<std::string>(device_class);
+    }).handle_exception([FNAME](std::exception_ptr e) {
+      ERROR("Failed to get device class: ", e);
+      return seastar::make_ready_future<std::string>("");
+    });
```

Comment:
Are we expecting an thrown exception here?
For expected errors in Crimson we use errorator, see dev/crimson/error-handling.

Reply:
I have changed the code now, please let me know if it is ok now.